lib/rule-tester/rule-tester.js

```diff
@@ -713,54 +861,28 @@ class RuleTester {
 				return calculatedConfig;
 			};
 
-			let code, filename, output, beforeAST, afterAST;
-
-			if (typeof item === "string") {
-				code = item;
-			} else {
-				code = item.code;
-
-				/*
-				 * Assumes everything on the item is a config except for the
-				 * parameters used by this tester
-				 */
-				const itemConfig = { ...item };
-
-				for (const parameter of RuleTesterParameters) {
-					delete itemConfig[parameter];
-				}
+			const { code, filename, options = [] } = item;
```

Comment:
`filename` and `options` were used only if they are test case object's own properties.

Reply:
Theoretically yes, but they seem to be very extreme examples that are unlikely to occur in real situations. Do you really think we need to consider this scenario?

Reply:
I think we could revisit this at some point, but the code looks like inherited properties are intentionally ignored.